src/java.base/share/classes/sun/security/util/DerOutputStream.java

```diff
@@ -461,8 +461,8 @@ public void putGeneralString(String s) throws IOException {
      * @param s the string to write
      * @param stringTag one of the DER string tags that indicate which
      * encoding should be used to write the string out.
-     * @param enc the name of the encoder that should be used corresponding
-     * to the above tag.
+     * @param charset the charset should be used corresponding to the
```

Comment:
I may suggest to add 'that' back.  Without it, the meaning of the sentence may be different.
```
... the charset should be used corresponding to the above tag.
vs
... the charset **that** should be used corresponding to the above tag.
```

Reply:
Thanks again for all the above suggestions sincerely!
To avoid adding your troubles, I should end up using the following expression:
```
the charset that should be used corresponding to the above tag.
```
I'm trying to understand the difference between the two.
The first sentence may be understood as "the process of using charset" according to the above label.
The second sentence(use that) can better express "the choice of charset" needs to be based on the above label.
Is this correct?

Reply:
> A spec for an argument should be a noun. You can add attributes to describe it in more detail, which can be an adjective, or even a clause. This is what's shown in the 2nd line (I cannot call it a sentence because it's just a noun with an attributive clause). On the other hand, the 1st is a full sentence and it's not what we need here.

Thanks so much for your detailed explanation.
I get it now.